metabrainz/admin/views.py

```diff
@@ -241,6 +245,54 @@ def overview(self):
             token_actions=TokenLog.list(10)[0],
         )
 
+    @staticmethod
+    def dns_lookup(ip):
+        try:
+            data = socket.gethostbyaddr(ip)
+            return data[0]
+        except Exception as err:
+            return None
+
+
+    @staticmethod
+    def lookup_ips(users):
+        """ Try to lookup and cache as many reverse DNS as possible in a window of time """
+
+        data = []
+        timeout = time.time() + StatsView.IP_ADDR_TIMEOUT
```

Comment:
if this is python3 you can use time.monotonic(): https://docs.python.org/3/library/time.html#time.monotonic